src/daft-dsl/src/lit.rs

```diff
@@ -246,6 +246,71 @@ impl LiteralValue {
             Python(..) => display_sql_err,
         }
     }
+
+    /// If the liter is a boolean, return it. Otherwise, return None.
+    pub fn as_bool(&self) -> Option<bool> {
```

Comment:
Would it be worth writing a macro for all these `as_` functions? Seems like a good amount of boilerplate

Reply:
some of them such as primitives (bool, i32, ...) dereference, but others return a reference, so I think the extra boilerplate is likely needed.